jaxrs-client/src/main/java/io/micronaut/jaxrs/client/JaxRsInvocation.java

```diff
@@ -0,0 +1,525 @@
+                            if (isResponseReturn()) {
+                                try {
+                                    MutableHttpResponse<?> mutableResponse = response.toMutableResponse();
+                                    JaxRsMutableResponse jaxRsMutableResponse = filterResponse(mutableResponse, requestContext);
+                                    complete(jaxRsMutableResponse);
+                                } catch (Exception e) {
+                                    future.completeExceptionally(new ProcessingException(e));
+                                }
+                            } else {
+                                future.completeExceptionally(new WebApplicationException(new JaxRsResponse(response)));
+                            }
+                        } else {
+                            future.completeExceptionally(new ProcessingException(throwable));
+                        }
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        if (!future.isDone()) {
+                            future.completeExceptionally(new ProcessingException("Expected a response"));
+                        }
+                    }
+
+                    private void complete(JaxRsMutableResponse jaxRsMutableResponse) {
+                        if (isResponseReturn()) {
+                            future.complete((T) jaxRsMutableResponse);
+                        } else {
+                            future.complete(jaxRsMutableResponse.readEntity(type));
+                        }
+                    }
+
+                    private boolean isResponseReturn() {
+                        return type.getType().equals(Response.class);
+                    }
+                });
+        } catch (Exception e) {
+            future.completeExceptionally(new ProcessingException(e));
+        }
+        return future;
+    }
+
+    private Response filterResponse(Response response, JaxRsClientRequestContext requestContext) {
+        if (response instanceof JaxRsMutableResponse jaxRsMutableResponse) {
+            jaxRsMutableResponse = jaxRsMutableResponse.withEntityReader(configuration.createHttpMessageEntityReader());
+            filterResponse(jaxRsMutableResponse, requestContext);
+            return jaxRsMutableResponse;
+        }
+        if (response instanceof JaxRsResponse jaxRsResponse) {
+            jaxRsResponse = jaxRsResponse.withEntityReader(configuration.createHttpMessageEntityReader());
+            MutableHttpResponse<?> mutableResponse = jaxRsResponse.getResponse().toMutableResponse();
+            JaxRsMutableResponse jaxRsMutableResponse = new JaxRsMutableResponse(mutableResponse, configuration.createHttpMessageEntityReader());
+            filterResponse(jaxRsMutableResponse, requestContext);
+            return jaxRsMutableResponse;
+        }
+        // Todo: filter custom response
```

Comment:
what about this

Reply:
Not sure if at this point we can have a custom response